Restore backward compat' for initSynonymBrowser

julienbourdeau opened the pull request:
`initSynonymBrowser` was renamed `initSynonymIterator` few weeks back, unfortunately backward compat' was not preserved: https://github.com/algolia/algoliasearch-client-php/commit/f2681ea9629581857ed0feca3073977f490c49dc

This pull request aims to restore backward compatibility.

rayrutjes commented on src/AlgoliaSearch/Index.php:
```diff
@@ -1633,6 +1633,16 @@ public function initSynonymIterator($batchSize = 1000)
+    /**
+     * @deprecated Please use initSynonymIterator instead
+     * @param int $batchSize
+     * @return SynonymIterator
```
Previous version returned a `SynonymBrowser` instance, I guess that one should be brought back as well.

rayrutjes commented on tests/AlgoliaSearch/Tests/SynonymsExportTest.php:
```diff
@@ -65,6 +65,14 @@ public function testCanGetCurrentSynonymOfNewIterator()
         ), $synonym);
+
+        // Test for backward compatibility
+        $synonym = $this->index->initSynonymBrowser()->current();
```
Add an assertion checking the returned instance equals the previous class.

julienbourdeau commented:
Sorry guys, false alarm 🎉 

the SynonymBrowser was introduced and renamed between 1.21.0 and 1.22.0! It was never released, therefor there is no BC break. The error was in the doc, I have renamed the function and it should be live soon.
